src/block.h

```diff
@@ -7,10 +7,15 @@ enum block_type {
 };
 
 enum surface_type {
-    surface_none,
-    surface_wall,
-    surface_grate,
-    surface_text,
+    surface_blocks_light = 0x40,
+    surface_blocks_air = 0x80,
+    surface_none = 0,
+
+    surface_wall = surface_blocks_light | surface_blocks_air,
```

Comment:
will just need to add something more once we have more surfaces that can block air and/or light (using lower bits).